styx-scheduler-service/src/main/java/com/spotify/styx/flyte/FlyteAdminClientRunner.java

```diff
@@ -152,7 +209,95 @@ public void poll(FlyteExecutionId flyteExecutionId, RunState runState)
     } catch (Exception e) {
       throw new PollingException(flyteExecutionId, e);
     }
+  }
 
+  void init() {
+    scheduleWithJitter(this::terminateDanglingFlyteExecutions, scheduledExecutor, terminateDanglingFlyteExecInterval);
+  }
+
+  @VisibleForTesting
+  void terminateDanglingFlyteExecutions() {
+    try {
+      // TODO: add tracing
+      tryTerminateDanglingFlyteExecutions();
+    } catch (Throwable t) {
+      LOG.warn("Error while terminating dangling flyte executions", t);
+    }
+
+  }
+
+  private void tryTerminateDanglingFlyteExecutions() {
+    var projects = flyteAdminClient.listProjects();
+
+    var allFutures = new ArrayList<CompletableFuture<Void>>();
+    for (var project : projects.getProjectsList()) {
```

Comment:
Do we want to get all executions every time? I think it is going to be a huge when the system is running a while.

Reply:
I was thinking trying out filters to see if I can retrieve only running executions with styx annotations. But I was planning to do that in a different PR.

Other than that, what is your proposal? just doing a few projects every "tick"? we will need to keep state of what we have done and what is still missing. 

On the other hand, we already [iterate over all running Pods](https://github.com/spotify/styx/blob/6a79ce45715076074123ca5662458077bbe6a939/styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesDockerRunner.java#L568) in KubernetesRunner.

Reply:
The executions are a bit different than Pods as k8s Pods will be cleaned up but executions will not so it will be accumulated. 

I think if the executions we are getting are sorted by time order, we can keep track the time of last execution that has been cleaned up and we don't need to going further in history.

Reply:
So for each flyte workflow we keep track of the last clean up execution time and update it if we cleanup some executions. What do you think?

Reply:
Interesting idea, do we need to keep a time for every workflow? wouldn't last cleanup date would be enough?

Reply:
We talk in a video call and we agreed to reduce the number of execution retrieved from Flyte in a different PR.
As a first alternative would be to leverage filters to see if we can get only running executions or in the worst case execution not older than x time (1.5 days maybe).